### PR TITLE
Add classifier component to app-project

### LINKED ISSUE #390: Add classifier library to app-project
**Package**

app-project

**Feature or Issue Description**

Hook up the classifier component to app-project, and ensure auth, project etc. props are correctly passed in

### PR DESCRIPTION
**Pull request by rogerhutchings**
Package: app-project

Adds the lib-classifier component to the project app and gets it to play ball. 

**Note:** The classifier doesn't render anything, since at this commit it appears to be broken -- or at least, was not working when run standalone either -- and is probably waiting on some of @shaunanoordin's PRs.

Close #390
Close #300
Close #345

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

**Comment by rogerhutchings** [comment]
Okay, it looks like the project is being loaded into the classifier component, but it's not fetching subjects / workflows

**Comment by eatyourgreens** [comment]
@rogerhutchings does the project have a default workflow, or any linked workflows? I'm looking at one in the dev tools and I can't see either of those on the project.

**Comment by rogerhutchings** [comment]
No, I found the issue - the project store in the app only stores the resource properties that it needs, not the properties that the project store in the classifier lib needs. I've fixed that and am fixing up some auth bits, which _should_ do it...

Side note - this would be another argument for #406

**Comment by rogerhutchings** [comment]
I think it should work 🎉 

You'll need to add the `?env=staging` parameter to the URL, because for some reason the classifier is hitting the live endpoint by default. But I think a fix for _that_ is outside the scope of this PR.

@eatyourgreens would you mind taking a quick look?

**Comment by eatyourgreens** [comment]
Ignore my comments about auth. It works if you run the app on a zooniverse.org domain.

File: packages/app-project/stores/Project.js [hunk 1]
```diff
@@ -35,16 +36,23 @@ const Project = types
...
+            'id',
+            'links',
+            'retired_subjects_count',
+            'slug',
+            'subjects_count',
+            'urls',
+          ]
+
+          properties.forEach(property => self[property] = project[property])
+          console.info('self')
```
Comment (eatyourgreens) [inline packages/app-project/stores/Project.js]: Stray debugging code?

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/SubjectViewer.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ import React from 'react'
 import getViewer from './helpers/getViewer'
 
 function storeMapper (stores) {
+  console.info(stores.classifierStore.subjects.toJSON())
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/SubjectViewer.js]: Stray debugging code?

File: packages/lib-classifier/src/store/ResourceStore.js [hunk 3]
```diff
@@ -33,13 +33,19 @@ const ResourceStore = types
...
+      }
     },
 
-    setActive: flow(function * setActive (id) {
-      // console.info('setActive', id)
-      const active = self.resources.get(id) || null
-
+    setActive: flow(function * setActive (id = null) {
+      const active = self.resources.get(id)
+      console.info('new', active)
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/ResourceStore.js]: Stray debugging code?

File: packages/lib-classifier/src/store/UserProjectPreferencesStore.js [hunk 4]
```diff
@@ -28,22 +28,40 @@ const UserProjectPreferencesStore = types
       addDisposer(self, projectDisposer)
     }
 
-    function createTempUPPOrFetchUPP () {
+    function createTempUPP () {
+      const tempUPP = UserProjectPreferences.create({ id: 'guestPreferencesDoNotPost', preferences: {} })
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/store/UserProjectPreferencesStore.js]: Do we still need this? PFE doesn't use it any more.

Comment (rogerhutchings) [reply packages/lib-classifier/src/store/UserProjectPreferencesStore.js]: I don't know - @srallen can probably speak to this one

File: packages/app-project/stores/Project.js (hunk 1 above)
Comment (rogerhutchings) [reply packages/app-project/stores/Project.js]: removed

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/SubjectViewer.js (hunk 2 above)
Comment (rogerhutchings) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/SubjectViewer.js]: removed

File: packages/lib-classifier/src/store/ResourceStore.js (hunk 3 above)
Comment (rogerhutchings) [reply packages/lib-classifier/src/store/ResourceStore.js]: removed

**Pull request merged**